development/tests/test_server_requests.py

```diff
@@ -0,0 +1,40 @@
+from django.test import TestCase
+from unittest.mock import patch
+from development.server_requests import (
+    get_logs,
+    send_challenge,
+)
+
+
+def mocked_requests_get(*args, **kwargs):
+    data = dict()
+    for key, value in zip(kwargs['keys'], kwargs['values']):
+        data[key] = value
+
+    class MockResponse:
+        def __init__(self, json_data, status_code):
+            self.json_data = json_data
+            self.status_code = status_code
+
+        def json(self):
+            return self.json_data
+
+    return MockResponse(data, 200)
+
+
+class TestServerRequests(TestCase):
+
+    @patch(
+        'requests.get',
+        side_effect=lambda x: mocked_requests_get(
```

Comment:
```
import json
from unittest.mock import MagicMock

mocked_response = MagicMock(json=lambda x: json.loads(x), spec=["json"])

with patch('requests.get', return_value=mocked_respons) as request_get_mocked:
    ....
    request_get_mocked.assert_called_once_with(...)
```
I think you could simply use `return_value=mocked_response`

Reply:
done!